iati/home/tests/test_middleware.py

```diff
@@ -0,0 +1,55 @@
+"""A module of unit tests for redirects."""
+import pytest
+from django.conf import settings
+from factories import DocumentFactory
+from wagtail.core.models import Site, Page
+from about.models import AboutPage
+
+
+@pytest.mark.django_db
+class TestRedirectMiddleware():
+    """Tests for the custom redirect middleware."""
+
+    def create_document(self):
+        """Call the DocumentFactory to save a new document."""
+        return DocumentFactory.create()
+
+    def test_redirect_middleware_internal(self, client, rf):
+        """
+        Test behavior for internal redirects.
+
+        Create about page not dependent on default pages being added
+        via management command.
+        """
+        client_request = rf.get('/', follow=True)
+        site = Site.find_for_request(client_request)
```

Comment:
@Ocre42 
Checking that the host name of the request matches the Site host of the request - ensuring that the request is internal. 

These tests are to be executed in multiple places - locally, in docker in the future, on travis. For test reasons we must therefore ensure that the site host and request host are the same in order for the redirects to execute.